pkg/services/rpcsrv/params/txBuilder.go

```diff
@@ -13,82 +13,111 @@ import (
+func ExpandFuncParameterIntoScript(script *io.BinWriter, fp FuncParam) error {
+	switch fp.Type {
+	case smartcontract.ByteArrayType:
+		str, err := fp.Value.GetBytesBase64()
 		if err != nil {
 			return err
 		}
-		switch fp.Type {
-		case smartcontract.ByteArrayType:
-			str, err := fp.Value.GetBytesBase64()
-			if err != nil {
-				return err
-			}
-			emit.Bytes(script, str)
-		case smartcontract.SignatureType:
-			str, err := fp.Value.GetBytesBase64()
-			if err != nil {
-				return err
-			}
-			emit.Bytes(script, str)
-		case smartcontract.StringType:
-			str, err := fp.Value.GetString()
-			if err != nil {
-				return err
-			}
-			emit.String(script, str)
-		case smartcontract.Hash160Type:
-			hash, err := fp.Value.GetUint160FromHex()
-			if err != nil {
-				return err
-			}
-			emit.Bytes(script, hash.BytesBE())
-		case smartcontract.Hash256Type:
-			hash, err := fp.Value.GetUint256()
-			if err != nil {
-				return err
-			}
-			emit.Bytes(script, hash.BytesBE())
-		case smartcontract.PublicKeyType:
-			str, err := fp.Value.GetString()
-			if err != nil {
-				return err
-			}
-			key, err := keys.NewPublicKeyFromString(string(str))
-			if err != nil {
-				return err
-			}
-			emit.Bytes(script, key.Bytes())
-		case smartcontract.IntegerType:
-			bi, err := fp.Value.GetBigInt()
+		emit.Bytes(script, str)
+	case smartcontract.SignatureType:
+		str, err := fp.Value.GetBytesBase64()
+		if err != nil {
+			return err
+		}
+		emit.Bytes(script, str)
+	case smartcontract.StringType:
+		str, err := fp.Value.GetString()
+		if err != nil {
+			return err
+		}
+		emit.String(script, str)
+	case smartcontract.Hash160Type:
+		hash, err := fp.Value.GetUint160FromHex()
+		if err != nil {
+			return err
+		}
+		emit.Bytes(script, hash.BytesBE())
+	case smartcontract.Hash256Type:
+		hash, err := fp.Value.GetUint256()
+		if err != nil {
+			return err
+		}
+		emit.Bytes(script, hash.BytesBE())
+	case smartcontract.PublicKeyType:
+		str, err := fp.Value.GetString()
+		if err != nil {
+			return err
+		}
+		key, err := keys.NewPublicKeyFromString(string(str))
+		if err != nil {
+			return err
+		}
+		emit.Bytes(script, key.Bytes())
+	case smartcontract.IntegerType:
+		bi, err := fp.Value.GetBigInt()
+		if err != nil {
+			return err
+		}
+		emit.BigInt(script, bi)
+	case smartcontract.BoolType:
+		val, err := fp.Value.GetBoolean() // not GetBooleanStrict(), because that's the way C# code works
+		if err != nil {
+			return errors.New("not a bool")
+		}
+		emit.Bool(script, val)
+	case smartcontract.ArrayType:
+		val, err := fp.Value.GetArray()
+		if err != nil {
+			return err
+		}
+		err = ExpandArrayIntoScriptAndPack(script, val)
+		if err != nil {
+			return err
+		}
+	case smartcontract.AnyType:
+		if fp.Value.IsNull() || len(fp.Value.RawMessage) == 0 {
+			emit.Opcodes(script, opcode.PUSHNULL)
+		}
+	case smartcontract.MapType:
+		vals, err := fp.Value.GetArray()
+		if err != nil {
+			return err
+		}
+		for i := len(vals) - 1; i >= 0; i-- {
```

Comment:
Zero-sized slice should be handled via `NEWMAP` opcode, same way as `ExpandArrayIntoScriptAndPack` does for arrays.

Reply:
Fixed.